api/dimensions.go

```diff
@@ -23,50 +22,16 @@ var MaxIDs = func() int {
 	return maxIDs
 }
 
-func (api *DatasetAPI) getDimensions(w http.ResponseWriter, r *http.Request) {
+func (api *DatasetAPI) getDimensions(w http.ResponseWriter, r *http.Request, limit, offset int) (interface{}, int, error) {
```

Comment:
It'd be good to have named return values to help readability. Similarly for `getDimensionOptions`, `getVersions` and `getLists`

Reply:
Yes, this is a good idea. Thank you for spotting this.

Reply:
I know this is a private method but is there value in adding Godoc now to explain the returned parameters? `(interface{}, int, error)` is pretty vague so additional context would be useful - I'm thinking of future devs who aren't familiar with the pagination work.

Reply:
Thanks for getting the discussion going - adding the documentation definitely helped me 🙂